todo_app/templates/index.html

```diff
@@ -10,9 +10,31 @@ <h1 class="display-4">To-Do App</h1>
   <div class="row">
     <div class="col-md-12">
       <h2>Items</h2>
-      <ul class="list-group mb-4">
-        <!-- List the todo items here -->
-      </ul>
+    </div>
+  </div>
+  {% for task in tasks %}
+    {% if loop.index0 % 3 == 0 %}
+      <div class="row">
+    {% endif %}
+    <div class="col-md-4">
+      <div class="card" style="margin-bottom:20px">
+        <div class="card-body">
+          {{ task['title'] }}
+        </div>
+      </div>
+    </div>
+    {% if loop.index0 % 3 == 2 or loop.last %}
+      </div>
+    {% endif %}
+  {% endfor %}
```

Comment:
This is a nice solution for the three column layout, an alternative which I think is a little easier to process because it only has to care about the loop once ((stolen from bootstrap's guide)[https://getbootstrap.com/docs/4.1/layout/grid/#equal-width-multi-row]) might be:

```python suggestion
<div class="row">
    {% for task in tasks %}
      <div class="col-md-4">
        <div class="card" style="margin-bottom:20px">
          <div class="card-body">
            {{ task['title'] }}
          </div>
        </div>
      </div>
      {% if loop.index0 % 3 == 2 %}
        <div class="w-100"></div>
      {% endif %}
    {% endfor %}
  </div>
```

Reply:
I've extracted this out into an issue so that I can consider addressing it at some point in the future. Thanks!